cf-net/cf-net.c

```diff
@@ -0,0 +1,507 @@
+    {"manpage",     no_argument,        0, 'M'},
+    {"debug",       no_argument,        0, 'd'},
+    {"verbose",     no_argument,        0, 'v'},
+    {"inform",      no_argument,        0, 'I'},
+    {"dry-run",     no_argument,        0, 'n'},
+    {"version",     no_argument,        0, 'V'},
+    {"interactive", no_argument,        0, 'i'},
+    {"timestamp",   no_argument,        0, 'l'},
+    {"background",  optional_argument,  0, 'b'},
+    {"timeout",     required_argument,  0, 't'},
+    {NULL,          0,                  0, '\0'}
+};
+
+// TODO: Check that these are correct later
+static const char *const HINTS[] =
+{
+    "Print the help message",
+    "Print the man page",
+    "Enable debugging output",
+    "Enable verbose output",
+    "Enable basic information output",
+    "Do not perform action, only output what to do",
+    "Output the version of the software",
+    "Enable interactive mode for key trust",
+    "Log timestamps on each line of log output",
+    "Parallelize connections (50 by default)",
+    "Connection timeout, seconds",
+    NULL
+};
+
+// TODO: Consider global variables for performance and code style(?)
+typedef struct
+{
+    bool debug;
+    bool verbose;
+    bool inform;
+    bool dry_run;
+    bool version;
+    bool interactive;
+    bool timestamp;
+    int background;
+    int timeout;
+} CFNetOptions;
+
+void CFNetSetDefault(CFNetOptions *opts);
+int CFNetRun(CFNetOptions *opts, const char *CMD,char **args);
+int CFNetParse( int argc, char **argv,
+                CFNetOptions *opts, char **cmd, char ***args );
+
+int main(int argc, char **argv)
+{
+    CFNetOptions opts;
+    char *cmd = NULL;
+    char **args = NULL;
+
+    int ret = CFNetParse(   argc, argv,          // Inputs
+                            &opts, &cmd, &args); // Outputs
+    if (ret == -1)
+    {
+        exit(1);
+    }
+    ret = CFNetRun(&opts, cmd, args);
+    return ret;
+}
+
+void CFNetSetDefault(CFNetOptions *opts){
+    if (opts == NULL)
+    {
+        return;
+    }
+    opts->debug       = false;
+    opts->verbose     = false;
+    opts->inform      = false;
+    opts->dry_run     = false;
+    opts->version     = false;
+    opts->interactive = false;
+    opts->timestamp   = false;
+    opts->background  = 0;
+    opts->timeout     = 0;
+}
+
+int CFNetParse( int argc, char **argv,
+                CFNetOptions *opts, char **cmd, char ***args )
+{
+    extern char *optarg;
+    extern int optind;//, opterr, optopt;
+    if (opts == NULL)
+    {
+        return -1;
+    }
+    CFNetSetDefault(opts);
+    int c = 0;
+    int start_index = 1;
+    while ((c = getopt_long(    argc, argv, "hMdvInVilb:t:",
+                                OPTIONS, &start_index))
+            != -1)
+    {
+        switch (c)
+        {
+            case 'h':
+            {
+                Writer *w = FileWriter(stdout);
+                WriterWriteHelp(w, "cf-net", OPTIONS, HINTS, true);
```

Comment:
`cf-net` is having quite a complicated syntax, so the help message should tell me more, like the available commands and maybe print an example.